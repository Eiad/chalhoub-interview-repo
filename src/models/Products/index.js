import axios from "axios";

const Products = {
  all: () => {
    return axios.get("/data/shirtsData.json").then(result => {
      return result.data;
    });
  }
};

export default Products;
