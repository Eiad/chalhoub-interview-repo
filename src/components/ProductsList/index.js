import React, { useState, useEffect, useMemo } from "react";

import Products from "../../models/Products";
import ZoomImage from "../ZoomImage";

const ProductList = ({ data = [] }) => {
  const [indexToShow, setIndexToShow] = useState(0);
  const [products, setProducts] = useState(data);

  const product = useMemo(() => {
    const currentProduct = products[indexToShow];

    //to cater for loading initial state
    return currentProduct != null ? currentProduct : {};
  }, [products, indexToShow]);

  const goToIndex = index => {
    setIndexToShow(index);
  };

  useEffect(() => {
    Products.all().then(products => {
      setProducts(products);
    });
  }, []);

  return (
    <div>
      <div className="main-container">
        <header>
          <a href="#main" className="logo">
            <img src="https://i.imgur.com/TSBasZJ.jpg" alt="" />
          </a>
          <div className="thenav">
            <ul class="nav-ul section group">
              <li class="col span_1_of_4">
                <a href="#men">Men</a>
              </li>
              <li class="col span_1_of_4">
                <a href="#women">Women</a>
              </li>
              <li class="col span_1_of_4">
                <a href="#kids">Kids</a>
              </li>
              <li class="col span_1_of_4">
                <a href="#ftw">FTW</a>
              </li>
            </ul>
          </div>
          <div className="search-bar">
            <form action="">
              <input type="text" placeholder="Search..." />
              <button type="submit">
                <i className="fa fa-search" />
              </button>
            </form>
          </div>
        </header>
        <section className="main-body">
          <div className="main-product-info">
            <ZoomImage src={product.smallPicture} />
            <h4>
              {product.productTitle} <br />
              <small> by {product.brand}</small>
            </h4>
            <p>
              {product.price} {product.currency}
            </p>
          </div>
          <div className="sizes">
            <p>
              Available Sizes: <span>{product.availSize}</span> <br />
              Neck type: <span>{product.sleevType}</span>
            </p>
          </div>
          <div className={product.isActive}>
            {products.map((product, index) => {
              return (
                <button
                  className={product.color}
                  onClick={() => goToIndex(index)}
                />
              );
            })}
          </div>
          <button className="a2c">Add to cart</button>

          <div className="description">
            <h5>Product description</h5>
            <p>{product.descreption}</p>
          </div>

          <hr />
          <h5>Products Complement your selection</h5>

          <div className="related-products section group">
            <div className="col span_1_of_4">
              <img src={product.thumb1} alt="" />
            </div>
            <div className="col span_1_of_4">
              <img src={product.thumb2} alt="" />
            </div>
            <div className="col span_1_of_4">
              <img src={product.thumb3} alt="" />
            </div>
            <div className="col span_1_of_4">
              <img src={product.thumb4} alt="" />
            </div>
          </div>
        </section>
      </div>
    </div>
  );
};

export default ProductList;
