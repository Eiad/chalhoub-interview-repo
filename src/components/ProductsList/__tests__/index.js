import React from "react";
import renderer from "react-test-renderer";

import ProductsList from "../index";
import Products from "../../../models/Products";

describe("ProductsList", () => {
  test("snapshot renders", done => {
    Products.all().then(products => {
      const component = renderer.create(<ProductsList data={products} />);

      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();

      done();
    });
  });
});
