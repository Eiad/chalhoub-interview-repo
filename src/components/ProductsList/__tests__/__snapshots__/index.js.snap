// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList snapshot renders 1`] = `
<div>
  <div
    className="main-container"
  >
    <header>
      <a
        className="logo"
        href="#main"
      >
        <img
          alt=""
          src="https://i.imgur.com/TSBasZJ.jpg"
        />
      </a>
      <div
        className="thenav"
      >
        <ul
          class="nav-ul section group"
        >
          <li
            class="col span_1_of_4"
          >
            <a
              href="#men"
            >
              Men
            </a>
          </li>
          <li
            class="col span_1_of_4"
          >
            <a
              href="#women"
            >
              Women
            </a>
          </li>
          <li
            class="col span_1_of_4"
          >
            <a
              href="#kids"
            >
              Kids
            </a>
          </li>
          <li
            class="col span_1_of_4"
          >
            <a
              href="#ftw"
            >
              FTW
            </a>
          </li>
        </ul>
      </div>
      <div
        className="search-bar"
      >
        <form
          action=""
        >
          <input
            placeholder="Search..."
            type="text"
          />
          <button
            type="submit"
          >
            <i
              className="fa fa-search"
            />
          </button>
        </form>
      </div>
    </header>
    <section
      className="main-body"
    >
      <div
        className="main-product-info"
      >
        <div
          data-rmiz-wrap="visible"
          style={undefined}
        >
          <img
            alt=""
            src="https://i.imgur.com/Iyy2jVQ.jpg"
          />
          <button
            aria-label="Zoom image"
            data-rmiz-btn-open={true}
            onClick={[Function]}
          />
        </div>
        <h4>
          Tiger Shirt - Red
           
          <br />
          <small>
             by 
            Michael Kors
          </small>
        </h4>
        <p>
          250
           
          AED
        </p>
      </div>
      <div
        className="sizes"
      >
        <p>
          Available Sizes: 
          <span>
            S, M, L
          </span>
           
          <br />
          Neck type: 
          <span>
            Y Collar
          </span>
        </p>
      </div>
      <div
        className="colors red-border"
      >
        <button
          className="red"
          onClick={[Function]}
        />
        <button
          className="black"
          onClick={[Function]}
        />
        <button
          className="grey"
          onClick={[Function]}
        />
      </div>
      <button
        className="a2c"
      >
        Add to cart
      </button>
      <div
        className="description"
      >
        <h5>
          Product description
        </h5>
        <p>
          It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
        </p>
      </div>
      <hr />
      <h5>
        Products Complement your selection
      </h5>
      <div
        className="related-products section group"
      >
        <div
          className="col span_1_of_4"
        >
          <img
            alt=""
            src="https://i.imgur.com/6s7TCGu.jpg"
          />
        </div>
        <div
          className="col span_1_of_4"
        >
          <img
            alt=""
            src="https://i.imgur.com/JZUsesm.jpg"
          />
        </div>
        <div
          className="col span_1_of_4"
        >
          <img
            alt=""
            src="https://i.imgur.com/EBeCFad.jpg"
          />
        </div>
        <div
          className="col span_1_of_4"
        >
          <img
            alt=""
            src="https://i.imgur.com/ZVjfS2b.jpg"
          />
        </div>
      </div>
    </section>
  </div>
</div>
`;
